src/main/scala/tilelink/Broadcast.scala

```diff
@@ -74,72 +173,143 @@ class TLBroadcast(params: TLBroadcastParams)(implicit p: Parameters) extends Laz
+
+      /** Create the probe filter, to filter B channel and D channel transactions. */
+      val flatAddresses: Seq[AddressSet] = AddressSet.unify(edgeOut.manager.managers.flatMap(_.address))
+      val addressMask: BigInt = AddressDecoder(flatAddresses.map(Seq(_)), flatAddresses.map(_.mask).reduce(_|_))
+      /** create [[ProbeFilter]] instance. */
       val filter = Module(params.filterFactory(ProbeFilterParams(
         mshrs  = params.numTrackers,
         caches = caches.size,
         maxAddress  = edgeIn.manager.maxAddress,
         addressMask = addressMask & ~(params.lineBytes-1))))
 
-      // Create the request tracker queues
+      /** Create the request tracker queues. */
       val trackers = Seq.tabulate(params.numTrackers) { id =>
         Module(new TLBroadcastTracker(id, params.lineBytes, caches.size, params.bufferless, edgeIn, edgeOut)).io
       }
 
-      // We always accept E
+      /* Client E Channel:
+       * Always accept E.
+       */
       in.e.ready := true.B
+      /* Notify tracker received E of a transaction.
+       * In to D channel handling logic of client, MSHR(currently tracker ID) is encoded in the sink field,
+       * client will reply it in the sink.
+       * this decode sink from E channel, then notify correspond tracker.
+       */
       (trackers zip UIntToOH(in.e.bits.sink).asBools) foreach { case (tracker, select) =>
         tracker.e_last := select && in.e.fire()
       }
 
-      // Depending on the high source bits, we might transform D
+      /* Client D Channel:
+       * Forward or transform manger D channel based on the payload of the source.
+       */
+      /** Highest sourceId index, using endSourceId to detect. */
       val d_high = log2Ceil(edgeIn.client.endSourceId)
+      /** payload extracted from source. */
       val d_what = out.d.bits.source(d_high+1, d_high)
+      /** indicate this transaction should be dropped:
+        * Probe client -> client ProbeAckData -> Put manger(DROP) -> manger AccessAck(DROP)
+        */
       val d_drop = d_what === DROP
+      /** indicate this transaction has an available data field. */
       val d_hasData = edgeOut.hasData(out.d.bits)
+      /* D channel to response to clients. */
       val d_normal = Wire(in.d)
+      /** [[d_first]] is the update enable signal of [[d_trackerOH]].
+        * [[d_last]] tell tracker and filter the end of transaction in D channel.
+        */
       val (d_first, d_last, _) = edgeIn.firstlast(d_normal)
+      /** select tracker with:
+        * 1. tracker need d reply
+        * 2. tracker source is same with the client source.
+        *
+        * This is used for:
+        * 1. encode tracker id into sink of D, make client reply in E to this tracker.
+        * 2. filter will need this D.
+        */
       val d_trackerOH = VecInit(trackers.map { t => t.need_d && t.source === d_normal.bits.source }).asUInt holdUnless d_first
 
+      /** if d is valid, and not drop. */
```

Comment:
[Good comments don't repeat the code or explain it. They clarify its intent.](https://en.wikipedia.org/wiki/Comment_(computer_programming)#Code_description)  The next line of code literally says `!out.d.valid || !d_drop` so I don't think this comment adds any clarity.